modules/loggers/mod_log_config.c

```diff
@@ -1666,13 +1782,229 @@ static ap_log_writer *ap_log_set_writer(ap_log_writer *handle)
+                if (!lfd->portions[j]) {
+                    apr_json_object_set(sub_log_object,
+                                        attribute_name, APR_JSON_VALUE_STRING,
+                                        json_null, r->pool);
+                }
+                else if (!*lfd->portions[j]) {
+                    apr_json_object_set(sub_log_object,
+                                        attribute_name, APR_JSON_VALUE_STRING,
+                                        json_empty_string,
+                                        r->pool);
+                }
+                else {
+                    attribute_value = lfd->portions[j];
+                    apr_json_object_set(sub_log_object,
+                                        attribute_name, APR_JSON_VALUE_STRING,
+                                        apr_json_string_create(r->pool, attribute_value,
+                                                               APR_JSON_VALUE_STRING),
+                                        r->pool);
+                }
+            }
+
+            continue;
+        }
+
+        if (!lfd->portions[i]) {
+            apr_json_object_set(log_object,
+                                attribute_name, APR_JSON_VALUE_STRING,
+                                json_null, r->pool);
+
+        }
+        else if (!*lfd->portions[i]) {
+            apr_json_object_set(log_object,
+                                attribute_name, APR_JSON_VALUE_STRING,
+                                json_empty_string,
+                                r->pool);
+        }
+        else {
+            attribute_value = lfd->portions[i];
+            apr_json_object_set(log_object,
+                                attribute_name, APR_JSON_VALUE_STRING,
+                                apr_json_string_create(r->pool, attribute_value,
+                                                       APR_JSON_VALUE_STRING),
+                                r->pool);
+
+        }
+    }
+
+    ba = apr_bucket_alloc_create(r->pool);
+    bb = apr_brigade_create(r->pool, ba);
+
+    /* encode into bucket brigade */
+    apr_json_encode(bb, NULL, NULL, log_object, APR_JSON_FLAGS_WHITESPACE, r->pool);
+    apr_brigade_puts(bb, NULL, NULL, APR_EOL_STR);
+
+    /* TODO: detect overflow nvec */
+    apr_brigade_to_iovec(bb, vec, &nvec);
+
+    lfdj->portions = apr_palloc(r->pool, sizeof(char*) * nvec);
+    lfdj->lengths = apr_palloc(r->pool, sizeof(int) * nvec);
+    lfdj->total_len = 0;
+    for (i = 0; i < nvec; i++) {
+        lfdj->portions[i] = vec[i].iov_base;
+        lfdj->lengths[i] = vec[i].iov_len;
+        lfdj->total_len += vec[i].iov_len;
+    }
+    lfdj->nelts = nvec;
+    return lfdj;
+}
+#else
+static apr_size_t add_str(apr_array_header_t * ah_strs, apr_array_header_t * ah_strl, const char *str)
+{
+    char** strs = apr_array_push(ah_strs);
+    int* strl = apr_array_push(ah_strl);
+
+    *strs = (char*) str;
+    *strl = strlen(str);
```

Comment:
Worked around the segfault with:

`    *strl = str ? strlen(str) : 0;
`

Reply:
should be fixed now